src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileDebugTargetGenerator.cs

```diff
@@ -60,23 +60,23 @@ public override IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnu
             }
         }
 
-        private ILaunchSettingsProvider LaunchSettingProvider { get; }
+        private IVersionedLaunchSettingsProvider LaunchSettingProvider { get; }
         private IProjectThreadingService ProjectThreadingService { get; }
 
         /// <summary>
         /// This provides access to the class which creates the list of debugger values..
         /// </summary>
         public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
             => Task.FromResult<IDynamicEnumValuesGenerator>(
-                new DebugProfileEnumValuesGenerator(LaunchSettingProvider, ProjectThreadingService));
+                new DebugProfileEnumValuesGenerator((ILaunchSettingsProvider)LaunchSettingProvider, ProjectThreadingService));
```

Comment:
Did you consider having `IVersionedLaunchSettingsProvider` derive from `ILaunchSettingsProvider` to avoid this implicit contract between interfaces?

Reply:
Yes, and I'm still thinking about it. :-)

At that point both `ILaunchSettingsProvider` and `IVersionedLaunchSettingsProvider` would have members with the same name, so the definition and implementation might get messy. But if we make `IVersionedLaunchSettingsProvider` public that would really be the right way to go.

Reply:
Done. Note this just moved the cast elsewhere to clarify _which_ `SourceBlock` we wanted to link to.